File: logcal/Views/Components/TodaysActivityCard.swift
//
//  TodaysActivityCard.swift
//  logcal
//
//  Created by Antriksh Johri on 20/08/26.
//

import SwiftUI

struct TodaysActivityCard: View {
    @Environment(\.colorScheme) var colorScheme
    let activeBurn: Double
    let basalBurn: Double
    let consumedCalories: Double
    let steps: Int
    let workouts: [HealthWorkoutItem]
    
    @State private var showExplanationSheet = false
    
    private var totalBurn: Double {
        basalBurn + activeBurn
    }
    
    private var netBalance: Double {
        totalBurn - consumedCalories
    }
    
    private var formattedSteps: String {
        let formatter = NumberFormatter()
        formatter.numberStyle = .decimal
        return formatter.string(from: NSNumber(value: steps)) ?? "\(steps)"
    }
    
    var body: some View {
        VStack(alignment: .leading, spacing: 16) {
            // 1. Header with Info (i) Button
            HStack {
                HStack(spacing: 8) {
                    ZStack {
                        Circle()
                            .fill(Color.orange.opacity(colorScheme == .dark ? 0.25 : 0.12))
                            .frame(width: 32, height: 32)
                        
                        Image(systemName: "flame.fill")
                            .font(.system(size: 16))
                            .foregroundColor(.orange)
                    }
                    
                    VStack(alignment: .leading, spacing: 2) {
                        Text("Activity & Energy Balance")
                            .font(.system(size: 16, weight: .bold, design: .rounded))
                            .foregroundColor(Theme.primaryText(colorScheme: colorScheme))
                        
                        Text("Apple Health")
                            .font(.system(size: 11, weight: .semibold, design: .rounded))
                            .foregroundColor(Theme.mutedText(colorScheme: colorScheme))
                    }
                }
                
                Spacer()
                
                Button {
                    showExplanationSheet = true
                } label: {
                    Image(systemName: "info.circle")
                        .font(.system(size: 18, weight: .medium))
                        .foregroundColor(Theme.mutedText(colorScheme: colorScheme))
                        .padding(6)
                        .contentShape(Rectangle())
                }
                .buttonStyle(.plain)
            }
            
            // 2. Stats Grid (Active Calories, Total Calories (by midnight), Steps)
            HStack(spacing: 8) {
                // Tile 1: Active Calories
                VStack(alignment: .leading, spacing: 6) {
                    VStack(alignment: .leading, spacing: 1) {
                        Text("Active Calories")
                            .font(.system(size: 10.5, weight: .bold, design: .rounded))
                            .foregroundColor(Theme.mutedText(colorScheme: colorScheme))
                            .lineLimit(1)
                            .minimumScaleFactor(0.85)
                        Text(" ")
                            .font(.system(size: 9, weight: .medium, design: .rounded))
                    }
                    .frame(height: 26, alignment: .topLeading)
                    
                    HStack(alignment: .firstTextBaseline, spacing: 2) {
                        Text("\(Int(activeBurn))")
                            .font(.system(size: 19, weight: .bold, design: .rounded))
                            .foregroundColor(Theme.primaryText(colorScheme: colorScheme))
                        Text("cal")
                            .font(.system(size: 11, weight: .semibold, design: .rounded))
                            .foregroundColor(Theme.mutedText(colorScheme: colorScheme))
                    }
                }
                .frame(maxWidth: .infinity, alignment: .leading)
                .padding(10)
                .background(Theme.insetBackground(colorScheme: colorScheme))
                .cornerRadius(12)
                
                // Tile 2: Total Calories (by midnight)
                VStack(alignment: .leading, spacing: 6) {
                    VStack(alignment: .leading, spacing: 1) {
                        Text("Total Calories")
                            .font(.system(size: 10.5, weight: .bold, design: .rounded))
                            .foregroundColor(Theme.primaryText(colorScheme: colorScheme))
                            .lineLimit(1)
                            .minimumScaleFactor(0.85)
                        Text("(by midnight)")
                            .font(.system(size: 9, weight: .semibold, design: .rounded))
                            .foregroundColor(Theme.mutedText(colorScheme: colorScheme))
                            .lineLimit(1)
                            .minimumScaleFactor(0.8)
                    }
                    .frame(height: 26, alignment: .topLeading)
                    
                    HStack(alignment: .firstTextBaseline, spacing: 2) {
                        Text("\(Int(totalBurn))")
                            .font(.system(size: 19, weight: .bold, design: .rounded))
                            .foregroundColor(Theme.primaryText(colorScheme: colorScheme))
                        Text("cal")
                            .font(.system(size: 11, weight: .semibold, design: .rounded))
                            .foregroundColor(Theme.mutedText(colorScheme: colorScheme))
                    }
                }
                .frame(maxWidth: .infinity, alignment: .leading)
                .padding(10)
                .background(Theme.insetBackground(colorScheme: colorScheme))
                .cornerRadius(12)
                
                // Tile 3: Steps
                VStack(alignment: .leading, spacing: 6) {
                    VStack(alignment: .leading, spacing: 1) {
                        Text("Steps")
                            .font(.system(size: 10.5, weight: .bold, design: .rounded))
                            .foregroundColor(Theme.mutedText(colorScheme: colorScheme))
                            .lineLimit(1)
                        Text(" ")
                            .font(.system(size: 9, weight: .medium, design: .rounded))
                    }
                    .frame(height: 26, alignment: .topLeading)
                    
                    HStack(alignment: .firstTextBaseline, spacing: 2) {
                        Text(formattedSteps)
                            .font(.system(size: 19, weight: .bold, design: .rounded))
                            .foregroundColor(Theme.primaryText(colorScheme: colorScheme))
                        Text("st")
                            .font(.system(size: 11, weight: .semibold, design: .rounded))
                            .foregroundColor(Theme.mutedText(colorScheme: colorScheme))
                    }
                }
                .frame(maxWidth: .infinity, alignment: .leading)
                .padding(10)
                .background(Theme.insetBackground(colorScheme: colorScheme))
                .cornerRadius(12)
            }
            
            // 3. Estimated Net Calorie Balance (24-Hour Projection)
            if totalBurn > 0 {
                VStack(alignment: .leading, spacing: 8) {
                    HStack {
                        if netBalance >= 0 {
                            HStack(spacing: 6) {
                                Image(systemName: "checkmark.circle.fill")
                                    .font(.system(size: 13))
                                    .foregroundColor(Theme.primaryGreen)
                                Text("Est. \(Int(netBalance)) kcal Deficit")
                                    .font(.system(size: 14, weight: .bold, design: .rounded))
                                    .foregroundColor(Theme.primaryGreen)
                            }
                        } else {
                            HStack(spacing: 6) {
                                Image(systemName: "bolt.fill")
                                    .font(.system(size: 13))
                                    .foregroundColor(Color.orange)
                                Text("Est. \(Int(-netBalance)) kcal Surplus")
                                    .font(.system(size: 14, weight: .bold, design: .rounded))
                                    .foregroundColor(Color.orange)
                            }
                        }
                        
                        Spacer()
                        
                        Text("Est. Burn \(Int(totalBurn)) • Eaten \(Int(consumedCalories))")
                            .font(.system(size: 11, weight: .medium, design: .rounded))
                            .foregroundColor(Theme.mutedText(colorScheme: colorScheme))
                    }
                    
                    // Visual Balance Progress Bar
                    GeometryReader { geo in
                        let maxVal = max(totalBurn, consumedCalories, 1.0)
                        let burnedRatio = min(1.0, totalBurn / maxVal)
                        let eatenRatio = min(1.0, consumedCalories / maxVal)
                        
                        ZStack(alignment: .leading) {
                            // Background track
                            Capsule()
                                .fill(Theme.insetBackground(colorScheme: colorScheme))
                                .frame(height: 8)
                            
                            // Burned bar
                            Capsule()
                                .fill(Color.orange.opacity(0.4))
                                .frame(width: geo.size.width * burnedRatio, height: 8)
                            
                            // Eaten bar
                            Capsule()
                                .fill(netBalance >= 0 ? Theme.primaryGreen : Color.orange)
                                .frame(width: geo.size.width * eatenRatio, height: 8)
                        }
                    }
                    .frame(height: 8)
                }
                .padding(12)
                .background(Theme.insetBackground(colorScheme: colorScheme))
                .cornerRadius(12)
            }
            
            // 4. Workouts Section
            VStack(alignment: .leading, spacing: 8) {
                Text(workouts.isEmpty ? "WORKOUTS" : "WORKOUTS (\(workouts.count))")
                    .font(.system(size: 11, weight: .bold, design: .rounded))
                    .foregroundColor(Theme.mutedText(colorScheme: colorScheme))
                    .padding(.top, 4)
                
                if workouts.isEmpty {
                    HStack(spacing: 6) {
                        Image(systemName: "figure.walk")
                            .font(.system(size: 13))
                            .foregroundColor(Theme.mutedText(colorScheme: colorScheme))
                        Text("No workouts recorded for this day")
                            .font(.system(size: 13, weight: .medium, design: .rounded))
                            .foregroundColor(Theme.mutedText(colorScheme: colorScheme))
                    }
                    .padding(.vertical, 2)
                } else {
                    VStack(spacing: 8) {
                        ForEach(workouts) { workout in
                            HStack(spacing: 12) {
                                ZStack {
                                    Circle()
                                        .fill(Theme.softAccentBackground(colorScheme: colorScheme))
                                        .frame(width: 36, height: 36)
                                    
                                    Image(systemName: workout.iconName)
                                        .font(.system(size: 16, weight: .semibold))
                                        .foregroundColor(Theme.primaryGreen)
                                }
                                
                                VStack(alignment: .leading, spacing: 2) {
                                    Text(workout.title)
                                        .font(.system(size: 14, weight: .bold, design: .rounded))
                                        .foregroundColor(Theme.primaryText(colorScheme: colorScheme))
                                    
                                    Text("\(formattedTime(workout.startDate)) • \(workout.durationMinutes) min")
                                        .font(.system(size: 12, weight: .medium, design: .rounded))
                                        .foregroundColor(Theme.mutedText(colorScheme: colorScheme))
                                }
                                
                                Spacer()
                                
                                if workout.caloriesBurned > 0 {
                                    HStack(spacing: 4) {
                                        Image(systemName: "flame.fill")
                                            .font(.system(size: 10))
                                            .foregroundColor(.orange)
                                        Text("\(Int(workout.caloriesBurned)) cal")
                                            .font(.system(size: 12, weight: .bold, design: .rounded))
                                            .foregroundColor(.orange)
                                    }
                                    .padding(.horizontal, 8)
                                    .padding(.vertical, 4)
                                    .background(Color.orange.opacity(colorScheme == .dark ? 0.2 : 0.12))
                                    .clipShape(Capsule())
                                }
                            }
                            .padding(10)
                            .background(Theme.insetBackground(colorScheme: colorScheme))
                            .cornerRadius(12)
                        }
                    }
                }
            }
        }
        .padding(20)
        .background(
            RoundedRectangle(cornerRadius: Constants.Sizes.cornerRadius)
                .fill(Theme.cardBackground(colorScheme: colorScheme))
        )
        .overlay(
            RoundedRectangle(cornerRadius: Constants.Sizes.cornerRadius)
                .stroke(Theme.cardBorder(colorScheme: colorScheme), lineWidth: Constants.Sizes.borderWidth)
        )
        .shadow(color: Theme.shadowColor(colorScheme: colorScheme), radius: 18, x: 0, y: 10)
        .sheet(isPresented: $showExplanationSheet) {
            ActivityEnergyExplanationSheet()
        }
    }
    
    private func formattedTime(_ date: Date) -> String {
        let formatter = DateFormatter()
        formatter.timeStyle = .short
        return formatter.string(from: date)
    }
}

// MARK: - Activity & Energy Explanation Sheet
struct ActivityEnergyExplanationSheet: View {
    @Environment(\.colorScheme) var colorScheme
    @Environment(\.dismiss) var dismiss
    
    var body: some View {
        NavigationStack {
            ScrollView {
                VStack(alignment: .leading, spacing: 20) {
                    // Header Banner
                    HStack(spacing: 12) {
                        ZStack {
                            Circle()
                                .fill(Color.orange.opacity(colorScheme == .dark ? 0.25 : 0.12))
                                .frame(width: 48, height: 48)
                            
                            Image(systemName: "flame.fill")
                                .font(.system(size: 24))
                                .foregroundColor(.orange)
                        }
                        
                        VStack(alignment: .leading, spacing: 2) {
                            Text("Activity & Energy Guide")
                                .font(.system(size: 18, weight: .bold, design: .rounded))
                                .foregroundColor(Theme.primaryText(colorScheme: colorScheme))
                            
                            Text("How your calorie expenditure is calculated")
                                .font(.system(size: 13, weight: .regular, design: .rounded))
                                .foregroundColor(Theme.mutedText(colorScheme: colorScheme))
                        }
                    }
                    .padding(.top, 8)
                    
                    Divider()
                    
                    // Item 1: Active Calories
                    explanationRow(
                        icon: "figure.run",
                        iconColor: .orange,
                        title: "Active Calories",
                        description: "Energy burned from physical movement, walking, workouts, and exercise sessions logged on Apple Watch or iPhone."
                    )
                    
                    // Item 2: Total Calories (by midnight / TDEE)
                    explanationRow(
                        icon: "bolt.fill",
                        iconColor: Color(red: 0.95, green: 0.70, blue: 0.25),
                        title: "Total Calories (by midnight)",
                        description: "Your estimated 24-hour Total Daily Energy Expenditure (TDEE). This combines your baseline Resting Metabolic Rate (BMR) with your active movement."
                    )
                    
                    // Item 3: Steps
                    explanationRow(
                        icon: "shoeprints.fill",
                        iconColor: Theme.primaryGreen,
                        title: "Daily Steps",
                        description: "Step count tracked continuously by your Apple Watch and iPhone pedometer."
                    )
                    
                    // Item 4: Est. Deficit / Surplus
                    explanationRow(
                        icon: "arrow.left.arrow.right",
                        iconColor: Theme.accentBlue,
                        title: "Estimated Deficit & Surplus",
                        description: "The estimated net difference between your 24-hour total energy burn and total calories eaten today. A calorie deficit supports fat loss, while a surplus supports muscle gain."
                    )
                    
                    // Item 5: Apple Health Privacy Note
                    HStack(alignment: .top, spacing: 10) {
                        Image(systemName: "lock.shield.fill")
                            .font(.system(size: 16))
                            .foregroundColor(Theme.mutedText(colorScheme: colorScheme))
                        
                        Text("All activity data is read directly and securely from Apple Health on your device.")
                            .font(.system(size: 12, weight: .regular, design: .rounded))
                            .foregroundColor(Theme.mutedText(colorScheme: colorScheme))
                    }
                    .padding(.top, 8)
                }
                .padding(20)
            }
            .background(Theme.backgroundColor(colorScheme: colorScheme).ignoresSafeArea())
            .navigationTitle("Energy & Activity")
            .navigationBarTitleDisplayMode(.inline)
            .toolbar {
                ToolbarItem(placement: .navigationBarTrailing) {
                    Button("Done") {
                        dismiss()
                    }
                    .font(.system(size: 15, weight: .bold, design: .rounded))
                    .foregroundColor(Theme.primaryGreen)
                }
            }
        }
        .presentationDetents([.medium, .large])
    }
    
    @ViewBuilder
    private func explanationRow(icon: String, iconColor: Color, title: String, description: String) -> some View {
        HStack(alignment: .top, spacing: 14) {
            ZStack {
                Circle()
                    .fill(iconColor.opacity(colorScheme == .dark ? 0.25 : 0.12))
                    .frame(width: 36, height: 36)
                
                Image(systemName: icon)
                    .font(.system(size: 16, weight: .semibold))
                    .foregroundColor(iconColor)
            }
            
            VStack(alignment: .leading, spacing: 4) {
                Text(title)
                    .font(.system(size: 15, weight: .bold, design: .rounded))
                    .foregroundColor(Theme.primaryText(colorScheme: colorScheme))
                
                Text(description)
                    .font(.system(size: 13, weight: .regular, design: .rounded))
                    .foregroundColor(Theme.mutedText(colorScheme: colorScheme))
                    .lineSpacing(2)
            }
        }
        .padding(12)
        .background(Theme.insetBackground(colorScheme: colorScheme))
        .cornerRadius(12)
    }
}

File: android/app/src/main/java/com/serene/logcal/ui/components/TodaysActivityCard.kt
package com.serene.logcal.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.navigationBarsPadding
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.DirectionsRun
import androidx.compose.material.icons.filled.FitnessCenter
import androidx.compose.material.icons.filled.Info
import androidx.compose.material.icons.filled.Lock
import androidx.compose.material.icons.filled.Speed
import androidx.compose.material.icons.filled.Whatshot
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.LinearProgressIndicator
import androidx.compose.material3.ModalBottomSheet
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.material3.rememberModalBottomSheetState
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.serene.logcal.service.HealthWorkout
import com.serene.logcal.ui.theme.LogCalTheme
import kotlin.math.absoluteValue
import kotlin.math.roundToInt

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun TodaysActivityCard(
    activeBurn: Double,
    basalBurn: Double,
    consumedCalories: Int,
    steps: Long,
    workouts: List<HealthWorkout>,
    modifier: Modifier = Modifier
) {
    val colors = LogCalTheme.colors
    var showInfoSheet by remember { mutableStateOf(false) }

    val totalBurn = (basalBurn + activeBurn).roundToInt()
    val netBalance = totalBurn - consumedCalories
    val isDeficit = netBalance >= 0

    val burnProgress = if (totalBurn > 0) {
        (consumedCalories.toFloat() / totalBurn.toFloat()).coerceIn(0f, 1.5f)
    } else 0f

    Box(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(16.dp))
            .background(colors.cardBackground)
            .border(1.dp, colors.cardBorder, RoundedCornerShape(16.dp))
            .padding(16.dp)
    ) {
        Column(modifier = Modifier.fillMaxWidth()) {
            // Header Row
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Row(verticalAlignment = Alignment.CenterVertically) {
                    Box(
                        modifier = Modifier
                            .size(34.dp)
                            .clip(CircleShape)
                            .background(Color(0xFF4DC6F5).copy(alpha = 0.15f)),
                        contentAlignment = Alignment.Center
                    ) {
                        Icon(
                            imageVector = Icons.AutoMirrored.Filled.DirectionsRun,
                            contentDescription = "Activity",
                            tint = Color(0xFF4DC6F5),
                            modifier = Modifier.size(18.dp)
                        )
                    }
                    Spacer(modifier = Modifier.width(10.dp))
                    Text(
                        text = "Activity & Energy",
                        fontSize = 17.sp,
                        fontWeight = FontWeight.Bold,
                        color = colors.primaryText
                    )
                }

                IconButton(
                    onClick = { showInfoSheet = true },
                    modifier = Modifier.size(28.dp)
                ) {
                    Icon(
                        imageVector = Icons.Default.Info,
                        contentDescription = "Info",
                        tint = colors.mutedText,
                        modifier = Modifier.size(18.dp)
                    )
                }
            }

            Spacer(modifier = Modifier.height(14.dp))

            // 3-Tile Stat Grid (compact, left-aligned)
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.spacedBy(8.dp)
            ) {
                // Tile 1: Active Calories
                StatTile(
                    title = "Active Calories",
                    value = "${activeBurn.roundToInt()}",
                    unit = "cal",
                    color = Color(0xFFF26161),
                    modifier = Modifier.weight(1f)
                )

                // Tile 2: Total Calories (by midnight)
                StatTile(
                    title = "Total (midnight)",
                    value = "$totalBurn",
                    unit = "cal",
                    color = Color(0xFFFFA500),
                    modifier = Modifier.weight(1f)
                )

                // Tile 3: Steps
                StatTile(
                    title = "Steps",
                    value = "$steps",
                    unit = "steps",
                    color = Color(0xFF4DC6F5),
                    modifier = Modifier.weight(1f)
                )
            }

            Spacer(modifier = Modifier.height(12.dp))

            // Estimated Net Deficit / Surplus Bar
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .clip(RoundedCornerShape(12.dp))
                    .background(colors.background)
                    .padding(12.dp)
            ) {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Text(
                        text = if (isDeficit) "Est. Net Deficit" else "Est. Net Surplus",
                        fontSize = 13.sp,
                        fontWeight = FontWeight.SemiBold,
                        color = colors.primaryText
                    )
                    Text(
                        text = "${netBalance.absoluteValue} cal ${if (isDeficit) "Deficit" else "Surplus"}",
                        fontSize = 13.sp,
                        fontWeight = FontWeight.Bold,
                        color = if (isDeficit) Color(0xFF4E9F3D) else Color(0xFFF26161)
                    )
                }

                Spacer(modifier = Modifier.height(8.dp))

                LinearProgressIndicator(
                    progress = { (burnProgress / 1.0f).coerceIn(0f, 1f) },
                    modifier = Modifier
                        .fillMaxWidth()
                        .height(8.dp)
                        .clip(RoundedCornerShape(4.dp)),
                    color = if (isDeficit) Color(0xFF4E9F3D) else Color(0xFFF26161),
                    trackColor = colors.cardBorder
                )
            }

            // Workouts List (if any)
            if (workouts.isNotEmpty()) {
                Spacer(modifier = Modifier.height(10.dp))
                Column(verticalArrangement = Arrangement.spacedBy(6.dp)) {
                    workouts.forEach { workout ->
                        Row(
                            modifier = Modifier
                                .fillMaxWidth()
                                .clip(RoundedCornerShape(10.dp))
                                .background(colors.background)
                                .padding(horizontal = 12.dp, vertical = 8.dp),
                            horizontalArrangement = Arrangement.SpaceBetween,
                            verticalAlignment = Alignment.CenterVertically
                        ) {
                            Row(verticalAlignment = Alignment.CenterVertically) {
                                Icon(
                                    imageVector = Icons.Default.FitnessCenter,
                                    contentDescription = "Workout",
                                    tint = Color(0xFF4DC6F5),
                                    modifier = Modifier.size(16.dp)
                                )
                                Spacer(modifier = Modifier.width(8.dp))
                                Text(
                                    text = workout.title,
                                    fontSize = 13.sp,
                                    fontWeight = FontWeight.Medium,
                                    color = colors.primaryText
                                )
                            }
                            Text(
                                text = "${workout.durationMinutes} min • ${workout.caloriesBurned.roundToInt()} cal",
                                fontSize = 12.sp,
                                fontWeight = FontWeight.Bold,
                                color = Color(0xFF4DC6F5)
                            )
                        }
                    }
                }
            }
        }
    }

    if (showInfoSheet) {
        ActivityExplanationBottomSheet(onDismiss = { showInfoSheet = false })
    }
}

@Composable
private fun StatTile(
    title: String,
    value: String,
    unit: String,
    color: Color,
    modifier: Modifier = Modifier
) {
    val colors = LogCalTheme.colors

    Column(
        modifier = modifier
            .clip(RoundedCornerShape(10.dp))
            .background(colors.background)
            .padding(horizontal = 10.dp, vertical = 8.dp),
        horizontalAlignment = Alignment.Start
    ) {
        Text(
            text = title,
            fontSize = 11.sp,
            fontWeight = FontWeight.Medium,
            color = colors.mutedText,
            maxLines = 1
        )
        Spacer(modifier = Modifier.height(2.dp))
        Row(verticalAlignment = Alignment.Bottom) {
            Text(
                text = value,
                fontSize = 15.sp,
                fontWeight = FontWeight.Bold,
                color = color
            )
            Spacer(modifier = Modifier.width(3.dp))
            Text(
                text = unit,
                fontSize = 10.sp,
                color = colors.mutedText,
                modifier = Modifier.padding(bottom = 1.dp)
            )
        }
    }
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
private fun ActivityExplanationBottomSheet(onDismiss: () -> Unit) {
    val colors = LogCalTheme.colors
    val sheetState = rememberModalBottomSheetState(skipPartiallyExpanded = true)

    ModalBottomSheet(
        onDismissRequest = onDismiss,
        sheetState = sheetState,
        containerColor = colors.background
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .navigationBarsPadding()
                .padding(horizontal = 20.dp, vertical = 16.dp)
        ) {
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(
                    text = "Activity & Energy Explained",
                    fontSize = 18.sp,
                    fontWeight = FontWeight.Bold,
                    color = colors.primaryText
                )
                TextButton(onClick = onDismiss) {
                    Text(text = "Done", fontWeight = FontWeight.Bold, color = Color(0xFF4E9F3D))
                }
            }

            Spacer(modifier = Modifier.height(12.dp))

            ExplanationItem(
                icon = Icons.Default.Whatshot,
                iconColor = Color(0xFFF26161),
                title = "Active Calories",
                description = "Calories burned through active movement, exercise sessions, and workouts recorded in Google Health Connect."
            )

            Spacer(modifier = Modifier.height(12.dp))

            ExplanationItem(
                icon = Icons.Default.Speed,
                iconColor = Color(0xFFFFA500),
                title = "Total Calories (by midnight)",
                description = "Your Total Daily Energy Expenditure (TDEE). Combines your resting BMR (what your body burns just staying alive all 24 hours) with your active burn."
            )

            Spacer(modifier = Modifier.height(12.dp))

            ExplanationItem(
                icon = Icons.AutoMirrored.Filled.DirectionsRun,
                iconColor = Color(0xFF4DC6F5),
                title = "Steps & Workouts",
                description = "Live step counts and workout sessions synced directly from connected devices (Pixel Watch, Galaxy Watch, etc.)."
            )

            Spacer(modifier = Modifier.height(12.dp))

            ExplanationItem(
                icon = Icons.Default.Lock,
                iconColor = Color(0xFF4E9F3D),
                title = "Privacy First",
                description = "Health data remains strictly on your device and is never shared with third parties or stored unencrypted."
            )

            Spacer(modifier = Modifier.height(16.dp))
        }
    }
}

@Composable
private fun ExplanationItem(
    icon: ImageVector,
    iconColor: Color,
    title: String,
    description: String
) {
    val colors = LogCalTheme.colors

    Row(modifier = Modifier.fillMaxWidth(), verticalAlignment = Alignment.Top) {
        Box(
            modifier = Modifier
                .size(36.dp)
                .clip(CircleShape)
                .background(iconColor.copy(alpha = 0.15f)),
            contentAlignment = Alignment.Center
        ) {
            Icon(
                imageVector = icon,
                contentDescription = title,
                tint = iconColor,
                modifier = Modifier.size(18.dp)
            )
        }
        Spacer(modifier = Modifier.width(12.dp))
        Column(modifier = Modifier.weight(1f)) {
            Text(
                text = title,
                fontSize = 14.sp,
                fontWeight = FontWeight.Bold,
                color = colors.primaryText
            )
            Spacer(modifier = Modifier.height(2.dp))
            Text(
                text = description,
                fontSize = 12.sp,
                color = colors.mutedText,
                lineHeight = 16.sp
            )
        }
    }
}
